### DISCUSSION TITLE
Support zed code editor.

### DISCUSSION THREAD

**Discussion by salemkode** [0no-co/gql.tada#375] [Q&A]
In my last time I try using gql tada in zed, it work

**Comment by Blankeos** [comment]
Same. I use Zed :(

But apparently for some it does work: https://github.com/zed-industries/zed/issues/12567. I think I just need a repro to figure it out.

```sh
bun gql-tada doctor
```

Seems to be okay for me, except VSCode extension (for obvious reasons).

<img width="679" alt="image" src="https://github.com/user-attachments/assets/232d3741-b8ba-4b76-9c5b-f8ed30b3ecad" />


Okay I got it working with a CLI command, which isn't that bad. Ideally the CLI will run automatically via the LSP, but this isn't bad. I think it's an easier alternative to graphql-codegen still.
```sh
bun gql-tada generate-output
```

**Reply by Blankeos** [reply]
Ohh finally got it working!!

Requirements for Zed
1. Install gql-tada (based on docs)
2. Make sure you have the [GraphQL Extension for Zed](https://github.com/11bit/zed-extension-graphql).
3. Follow the instructions for enabling LSP:
```

# In the root of your project
# graphql.config.yml
schema: "http://localhost:8080/api/graphql" # You can also pass a path to a schema.graphql or an introspection endpoint like this.
documents: "src/**/*.{graphql,ts,tsx}" # The files where you want the graphql intellisense to work.

```

**Reply by kitten** [reply]
Just to point this out, this **isn't** making gql.tada and its TS plugin / TS server LSPs work, this looks like it's adding a completely unrelated LSP plugin, separate from TypeScript. Basically, all notes as to why using the VSCode GraphQL (non-syntax) plugin is a bad idea in conjunction with gql.tada apply to this too.

The keyword to search here is whether Zed is able to use the in-workspace TSServer
